app/docs/setup_code.tsx

```diff
@@ -38,15 +43,36 @@ export default class SetupCodeComponent extends React.Component {
     });
 
     if (this.props.bazelConfigResponse) {
+      this.setConfigResponse(this.props.bazelConfigResponse);
       return;
     }
     let request = new bazel_config.GetBazelConfigRequest();
     request.host = window.location.host;
     request.protocol = window.location.protocol;
     request.includeCertificate = true;
-    rpcService.service.getBazelConfig(request).then((response: bazel_config.GetBazelConfigResponse) => {
-      this.setState({ ...this.state, bazelConfigResponse: response });
-    });
+    rpcService.service.getBazelConfig(request).then(this.setConfigResponse.bind(this));
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.bazelConfigResponse !== prevProps.bazelConfigResponse) {
+      this.setConfigResponse(this.props.bazelConfigResponse);
+    }
+  }
+
+  setConfigResponse(response: bazel_config.GetBazelConfigResponse) {
+    if (!response) {
```

Comment:
think you can remove this early exit if below you change response.credential to response?.credential